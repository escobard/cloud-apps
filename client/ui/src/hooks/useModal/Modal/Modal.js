import React from "react";
import PropTypes from "prop-types";
import { Modal, Grid } from "semantic-ui-react";

import "./Modal.scss";

const ModalElement = ({ title, content, showModal, closeModal }) => (
  <Modal id="modal" size="large" open={showModal} onClose={closeModal}>
    <Modal.Content>
      <Grid className="header">
        <Grid.Column width={2}>
          <i
            onKeyDown={e => {
              e.key === "Enter" && closeModal();
            }}
            tabIndex={0}
            aria-label="Back"
            onClick={closeModal}
            className="arrow left big icon"
            role="button"
          />
        </Grid.Column>
        <Grid.Column width={12}>
          <h2>{title}</h2>
        </Grid.Column>
        <Grid.Column width={2}>
          <i aria-hidden="true" className="clipboard outline big icon" />
        </Grid.Column>
      </Grid>
      <div className="icon-container">
        <i
          aria-hidden="true"
          className="sticky note outline big circular icon"
        />
      </div>
      {content}
    </Modal.Content>
  </Modal>
);

ModalElement.propTypes = {
  title: PropTypes.string.isRequired,
  content: PropTypes.shape({}).isRequired,
  showModal: PropTypes.bool.isRequired,
  closeModal: PropTypes.func.isRequired
};

export default ModalElement;
