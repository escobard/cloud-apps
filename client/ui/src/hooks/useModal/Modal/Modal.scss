@import "../../../styles/variables";

#modal{
  background: $blue-dark !important;
  > .content{
    background: $blue-dark !important;
    color: $white;
    .ui.message{
      &.red{
        .header {
          color: $red;
        }
      }
      &.green{
        .header {
          color: $green;
        }
      }

    }
    .header{
      color: $white;
      h2{
        text-align: center;
        font-weight: 100;
      }
      .arrow{
        &:hover{
          cursor: pointer;
        }
      }
      .clipboard{
        float: right;
      }
    }
    .icon-container{
      padding: $space-xxlrg 0 $space-mega;
      i{
        font-size: $font-size-xxxl;
        text-align: center;
        float: none;
        margin: auto;
        display: block;
        box-shadow: none;
        border: solid 2px $white;
      }
    }
    button{
      width: 100%;
      text-transform: capitalize;
      font-size: $font-size-small;
      color: $white;
      background: $blue;
      &:hover {
        background: $white;
        color: $blue;
      }
    }
    }
  }
