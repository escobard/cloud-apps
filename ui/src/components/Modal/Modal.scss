@import "../../styles/variables";

div[id*="-modal"]{
  background: $blue-dark !important;
  .content{
    background: $blue-dark !important;
    color: $white;
  }
  h2{
    text-align: center;
    font-weight: 100;
  }
  .clipboard{
    float: right;
  }
}