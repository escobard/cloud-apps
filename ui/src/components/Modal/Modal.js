import React from "react";
import { Modal, Grid } from "semantic-ui-react";
import PropTypes from "prop-types";

import "./Modal.scss";

/** Dynamically creates a modal from the data provided via props
 * @dev renders a semantic-ui-react modal
 * @param {boolean} state, determines open / close state
 * @param {function} close, state handler from parent
 * @param {string} title, string for modal title
 * @param {function} content, component to render in modal
 * @returns {Component}, ModalElement
 * */

const ModalElement = ({ state, close, data: {title, content}}) => (
  <Modal id="modal" size="large" open={state} onClose={close}>
    <Modal.Content>
      <Grid className="header">
        <Grid.Column width={2}>
          <i
            onKeyDown={e => {
              e.key === "Enter" && close();
            }}
            tabIndex={0}
            aria-label="Back"
            onClick={close}
            className="arrow left big icon"
          />
        </Grid.Column>
        <Grid.Column width={12}>
          <h2>{title}</h2>
        </Grid.Column>
        <Grid.Column width={2}>
          <i aria-hidden="true" className="clipboard outline big icon" />
        </Grid.Column>
      </Grid>
      <div className="icon-container">
        <i
          aria-hidden="true"
          className="sticky note outline big circular icon"
        />
      </div>
      {content}
    </Modal.Content>
  </Modal>
);



export default ModalElement;
