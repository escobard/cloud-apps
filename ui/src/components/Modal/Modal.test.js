import Modal from "./index";

describe(">> Modal component", () => {
  it(">> snapshot is up to date", () => {
    const { container } = render(<Modal />);
    expect(container).toMatchSnapshot();
  });
});
