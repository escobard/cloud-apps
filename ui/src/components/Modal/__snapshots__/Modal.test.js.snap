// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>> Modal component >> snapshot is up to date 1`] = `<div />`;

exports[`>> render() >> snapshot is up to date 1`] = `<div />`;

exports[`Modal component >> render() >> snapshot is up to date 1`] = `<div />`;
